main.rb

```diff
@@ -0,0 +1,147 @@
+module Enumerable
+  # 1.my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+    for item in self
+      yield(item)
+    end
+  end
+
+  # 2.my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    index = 0
+    my_each do |item|
+      yield(item, index)
+      index += 1
+    end
+  end
+
+  # 3.my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    select_arr_result = []
+    each do |item|
+      select_arr_result << item if yield(item)
+    end
+    select_arr_result
+  end
+
+  # 4.my_all?
+  def my_all?(_parameter = nil)
+    length_array = size
+    true_counter1 = 0
+    true_counter = 0
+    # Check if there is block given
+    if block_given?
+      my_each do |x|
+        true_counter += 1 if yield(x)
+      end
+      true_counter == length_array
+    else
+      my_each do |element|
+        true_counter1 += 1 if !element.nil? == true
+      end
+      true_counter1 == length_array
+    end
+  end
+
+  # 5.my_any?
+  def my_any?(parameter = nil)
+    initial_value = false
+    if block_given?
+      my_each do |x|
+        initial_value = true if yield(x)
+      end
+      initial_value
+    else
+      case parameter
+      when Regexp
+        my_each { |x| return true if x =~ parameter }
+      else
+        my_each do |element|
+          initial_value = true unless element.nil?
+        end
+      end
+      initial_value
+    end
+  end
```

Comment:
## my_any?

_use the same variables in the previous examples_

- [ ] when no block or argument is given returns false if at least one of the collection is not true:

```ruby

    false_array.my_any? === false_array.any? => false

```

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer:

```ruby

    words.my_any?(Integer) === words.any?(Integer) => false

```

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern:

```ruby

    words.my_any?('cat') === words.any?('cat') => false

```